diff --git a/MODULE.bazel b/MODULE.bazel
index 9021627..eb2e792 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -88,3 +88,4 @@ index_integrities(
     ],
     uri = "https://snapshot.debian.org/archive/debian/20250201T023325Z",
 )
+bazel_dep(name = "rules_shell", version = "0.4.0")
diff --git a/apt/private/deb_download.bzl b/apt/private/deb_download.bzl
index 413a8d3..5e78ea4 100644
--- a/apt/private/deb_download.bzl
+++ b/apt/private/deb_download.bzl
@@ -56,6 +56,8 @@ def _get_input_data_with_sources(rctx):
     return input_data
 
 _INDEX_BUILD_TMPL = """
+load("@rules_shell//shell:sh_binary.bzl", "sh_binary")
+
 filegroup(
     name = "lockfile",
     srcs = ["lock.json"],
